docker/app/Dockerfile

```diff
@@ -0,0 +1,84 @@
+# syntax=docker/dockerfile:1
+ARG RUBY_VERSION

```

Comment:
Unlike the previous PR, `RUBY_VERSION` is no longer hardcoded. It is sourced from the Makefile via an environment variable and passed to the Docker image build through Docker Compose.